sips/sip-017/sip-017-schnorr.md

```diff
@@ -0,0 +1,120 @@
+# SIP-017 — Schnorr signatures
+
+## Preamble
+
+**SIP:** 017 (awaiting acceptance)
+
+**Title:** Schnorr signatures
+
+**Author:** Asteria [asteria@syvita.org](mailto:asteria@syvita.org)
+
+**Consideration:** Technical
+
+**Type:** Consensus
+
+**Layer:** Consensus (soft fork)
+
+**Status:** Draft
+
+**Created:** 19 December 2021
+
+**License:** BSD 2-Clause
+
+**Sign-off:** —
+
+## Abstract
+
+This SIP describes the addition of Schnorr signature structure and validation to the account model for the Stacks blockchain.
+
+## Introduction
+
+The Stacks blockchain signing system was initially built on ECDSA, similar to how the Bitcoin blockchain was. However, it is widely acknowledged that Schnorr is the better signing algorithm over ECDSA. The primary reason it wasn’t used in the Bitcoin blockchain for many years is that Schnorr was patented until 2008, the same year Bitcoin was invented. At the time, Satoshi Nakomoto decided that Schnorr signatures lacked the popularity and testing required to secure a system as critical as Bitcoin.
```

Comment:
> However, it is widely acknowledged that Schnorr is the better signing algorithm over ECDSA

By whom?  I don't doubt that many cryptographers might prefer a Schnorr signature schema over ECDSA, but can this text point to specific technical arguments for Schnorr over ECDSA?  Maybe there's something in here for further reading: https://en.wikipedia.org/wiki/Schnorr_signature#Security_argument

> The primary reason it wasn’t used in the Bitcoin blockchain for many years is that Schnorr was patented until 2008

Is there a source for this?  Bitcoin was released in 2009, so I'm surprised to learn that this was indeed the motivating reason for omitting Schnorr signatures.